src/Temporalio/Common/RetryPolicy.cs

```
@@ -65,7 +65,7 @@ internal static RetryPolicy FromProto(Api.Common.V1.RetryPolicy proto)
         {
             return new()
             {
-                InitialInterval = proto.InitialInterval.ToTimeSpan(),
+                InitialInterval = (proto.InitialInterval ?? new Duration()).ToTimeSpan(),
```

Can you explain this change a bit? Is this sometimes null from the server? What does the `InitialInterval` in this PR end up being when that happens?

---

Without this change, the test `Temporalio.Tests.Worker.WorkflowWorkerTests.ExecuteWorkflowAsync_SimpleActivity_ExecutesProperly(local=true)` hangs up indefinitely due to NullReferenceException on `InitialInterval`. The value is null because default-constructed `Api.Common.V1.RetryPolicy` has it as null (`Duration` is a message type). I am still trying to find the root cause why Core SDK uses default-constructed `RetryPolicy` here - we may have a bug in local activity start message somewhere. But even if we fixed that part, I think we still don't want `RetryPolicy.FromProto` to crash when `InitialInterval` is null for whatever reason.

In this case, `InitialInterval` is initialized to zero. Alternatively we could use the fields default value (1 second) but I'm not a fan of it - it would make it impossible to differentiate missing/malformed RetryPolicy and a properly initialized RetryPolicy with default interval. Third alternative is to make `FromProto` return null.

---

> I am still trying to find the root cause why Core SDK uses default-constructed RetryPolicy here - we may have a bug in local activity start message somewhere

I think it's worth understanding the situation where `initial_interval` is _ever_ null from Core because I think we have several SDKs that assume it is _never_ null from Core. I don't think a zero initial interval is even considered valid for a retry policy (not sure .NET validates, but I see such a validation in Python) and I wouldn't want to set it some default value that doesn't relate to anything real.

---

Regardless of Core behavior, should we validate and return null on failed validation here, or should we throw?

---

IMO we should throw if we get unexpected values (same as we have until now). I have never seen this line throw before and IMO we have to be able to make assumptions in dozens of places about nullability of proto fields (or lack thereof) without manually checking each time.

---

Restored original behavior. After https://github.com/temporalio/sdk-core/pull/1012, Core guarantees `initial_interval` is not null.